src/lib/bootloader/mbr_update.rb

```diff
@@ -183,8 +156,8 @@ def activatable_partitions(disk)
       disk.partitions.reject { |p| p.id.is?(:swap, :bios_boot) }
     end
 
-    def extended_partition(disk)
-      part = activatable_partitions(disk).find { |p| p.type == Y2Storage::PartitionType::EXTENDED }
+    def extended_partition(partition)
+      part = partition.siblings.find { |p| p.type.is?(:extended) }
```

Comment:
I would avoid using methods like `siblings` because the organization of the devicegraph can be surprising sometimes. In general, if there is a more "semantic" way of doing things (instead of using methods that traverse the graph at "low level"), that's by far preferred.

For example, something like `partition.partitionable.partitions`. Once again, if you thing is worth it, we can even add `Partition#other_partitions` or something like that.